### PR TITLE
Made addUnique/dropUnique migration utils idempodent

### PR DESCRIPTION
**Pull request by naz**
refs https://github.com/TryGhost/Ghost/pull/12598

- This changeset adds idepmotence to situations where unique constraint has to be dropped or added to the table
- Note '4.0/07-alter-unique-constraint-for-posts-slug.js` was migration that was effected by lack of idempotence

Note to reviewers, didn't find any better way to search for unique constraints then doing raw queries on `TABLE_CONSTRAINTS` in MySQL and `PRAGMA index_list` for SQLite. Let me know if you know any better ways.

### CODE REVIEW

**Review by allouis — APPROVED** [review]
Might wanna add some logging for the case that we _do_ add/drop the constraint!

**Comment by naz** [comment]
Very good point @allouis. Will add!

File: core/server/data/schema/commands.js [hunk 1]
```diff
@@ -85,10 +131,16 @@ function addUnique(tableName, columns, transaction) {
...
-    });
+async function dropUnique(tableName, columns, transaction) {
+    const hasUniqueConstraint = await hasUnique(tableName, columns, transaction);
+
+    if (hasUniqueConstraint) {
+        return (transaction || db.knex).schema.table(tableName, function (table) {
+            table.dropUnique(columns);
+        });
+    } else {
+        logging.warn(`Constraint for ${columns} does not exists for table ${tableName}`);
```
Comment (daniellockyer) [inline core/server/data/schema/commands.js:142]: `s/exists/exist/`

**Review by daniellockyer — APPROVED** [review]
Same point as @allouis + my small logging message nit, but LGTM 👍🏻

**Review by rshbhgrg — APPROVED** [review]

**Pull request merged**